src/dal/views.py

```diff
@@ -25,9 +26,16 @@ class ViewMixin(object):
 
     def dispatch(self, request, *args, **kwargs):
         """Set :py:attr:`forwarded` and :py:attr:`q`."""
-        self.forwarded = json.loads(request.GET.get('forward', '{}'))
-        if request.method == 'POST':
-            self.forwarded = json.loads(request.POST.get('forward', '{}'))
+        try:
+            self.forwarded = json.loads(request.GET.get('forward', '{}'))
+            if request.method == 'POST':
+                self.forwarded = json.loads(request.POST.get('forward', '{}'))
+
+            if not isinstance(self.forwarded, dict):
+                raise ValueError
+        except ValueError:
+            return HttpResponseBadRequest('Invalid JSON data')
```

Comment:
405 sounds useful, OPTIONS not so much, but if you're up for it go ahead, don't forget we'll need unit tests so you might not want to go through implementing OPTIONS for this reason.